stl/inc/system_error

```diff
@@ -520,16 +522,84 @@ public:
+//   <system_error> and would likely put borderline programs over the TLS slot count limit
+// * can't declare the error_category as an ordinary constexpr variable for most compilers, because error_category
+//   has a virtual destructor (TRANSITION, __cpp_constexpr_dynamic_alloc)
+// * can't declare the error_category as an ordinary non-constexpr variable even with a constexpr constructor, because
+//   the compiler will emit code to destroy it which invalidates its use in these DLL shutdown scenarios
+//
+// As a result, we are left a somewhat unfortunate workaround: We create an atomic<uintptr_t> array to store the
+//   error_category instance, test if the first atomic is nonzero (acquire), and if so, we know we have formed the
+//   instance and can return a reinterpreted pointer to that storage. If the first atomic is zero, we write all except
+//   the first atomic (relaxed), then write the first one as a store-release. (The non-first values are transferred
+//   to other threads in the release sequence).
+//
+// Acknowledged undefined and implementation defined behavior happening here:
+// * There is a data race when filling in the values other than the first atomic; this is OK on all hardware we target
+//   because the racing threads are all writing identical values.
+// * We are reaching into the memory layout atomic<uintptr_t>[N] and assuming we can interpret that as some other type.
+// * We are assuming that virtual functions are implemented with a vfptr located as the first member of an object.
+// (there are probably others)
+//
+// Inspecting the resulting assembly of any callers of _Immortalize_memcpy_image is recommended.
+//
+
+#if defined(_M_CEE_PURE)
+// /clr:pure doesn't ever do constant initialization, so rely on the CLR and magic statics
+
+#pragma warning(push)
+#pragma warning(disable : 4640) // construction of local static object is not thread-safe (/Wall)
+template <class _Ty>
+_NODISCARD const _Ty& _Immortalize_memcpy_image() noexcept {
+    /* MAGIC */ static _Immortalizer_impl<_Ty> _Static;
+    return _Static._Storage;
+}
+#pragma warning(pop)
+#elif defined(__cpp_constexpr_dynamic_alloc)
+template <class _Ty>
+_NODISCARD const _Ty& _Immortalize_memcpy_image() noexcept {
+    constexpr _Ty _Static;
+    return _Static._Storage;
+}
+#else
+template <class _Ty>
+_NODISCARD const _Ty& _Immortalize_memcpy_image() noexcept {
+    // return reference to a memcpy'd default-initialized _Ty
+    // pre: A default-initialized _Ty sets the first pointer-sized field to nonzero
+    constexpr size_t _Pointer_count = sizeof(_Ty) / sizeof(uintptr_t);
+    static atomic<uintptr_t> _Storage[_Pointer_count];
+    static_assert(sizeof(_Storage) == sizeof(_Ty), "Bad storage size");
+    static_assert(alignof(decltype(_Storage)) >= alignof(_Ty), "Bad alignment assumptions");
+    if (_Storage[0].load(_STD memory_order_acquire) != 0) {

```

Comment:
We're in `std` and `memory_order_acquire` is not a function, so this should be unqualified.